### PR TITLE
swap badge and use proper coverprofile

### PR DESCRIPTION
**Pull request by jbpratt**
Hopefully the runner launches for this and I can get it properly working.

### CODE REVIEW

**Comment by jbpratt** [comment]
@nicklaw5 Hmm, I'm not too sure about this one, give me a bit to get this working. Sorry about this.

**Comment by jbpratt** [comment]
Ah, https://github.com/mattn/goveralls/issues/140 the culprit

**Comment by jbpratt** [comment]
I'm not exactly sure how Coveralls works in general, does something need to be swapped on that side to satisfy these changes?

**Comment by nicklaw5** [comment]
I'm not sure. I need to head to work now, but I can take a look over the next day or so.

Don't stress about it though.

**Comment by jbpratt** [comment]
Hey, sorry for the radio silence. Is there anything I can do to help out on this one? I feel a bit bad for opening the can of worms.

**Comment by nicklaw5** [comment]
Na its fine - these things happen :slightly_smiling_face:

I've been pretty busy. I'll take a quick look now and see what's what.

**Comment by jbpratt** [comment]
Maybe @mattn can provide some insight, it is entirely possible that I just mis-configured something.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,13 +18,22 @@ jobs:
...
           go get github.com/mattn/goveralls
+          go install -v github.com/mattn/goveralls
       - name: Test
-        run: go test -v -parallel=10 -covermode=count -coverprofile=coverage.out
+        run: |
+          cd ~/go/src/github.com/${GITHUB_REPOSITORY}
+          go test -v -parallel=10 -covermode=count -coverprofile=coverage.out
       - name: Send coverage
         env:
           COVERALLS_TOKEN: ${{ secrets.COVERALLS_TOKEN }}
```
Comment (nicklaw5) [inline .github/workflows/ci.yml:36]: This needs to be:

```suggestion
          COVERALLS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by jbpratt** [comment]
Woo! Sorry about all this :face_with_head_bandage:

**Comment by nicklaw5** [comment]
Ok - not sure that solves our issue completely.

You're workflow now only checks out the `master` branch on build, rather than the branch associated with the PR:
```
- name: Check out code the Go-Way
  run: go get github.com/${GITHUB_REPOSITORY}
```

**Comment by jbpratt** [comment]
> Ok - not sure that solves our issue completely.
> 
> You're workflow now only checks out the `master` branch on build, rather than the branch associated with the PR:
> 
> ```
> - name: Check out code the Go-Way
>   run: go get github.com/${GITHUB_REPOSITORY}
> ```

This was only done initially since the repo does not contain `go.mod` or `dep` file, `gopath` needs to be assumed for older versions (iirc). We can try ripping that stuff out and see what happens

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -19,13 +18,22 @@ jobs:
         uses: actions/setup-go@v2
         with:
           go-version: ${{ matrix.go }}
+      - name: Check out code the Go-Way
+        run: go get github.com/${GITHUB_REPOSITORY}
       - name: Setup
         run: |
+          cd ~/go/src/github.com/${GITHUB_REPOSITORY}
+          go get -v -t -d ./...
```
Comment (nicklaw5) [inline .github/workflows/ci.yml:26]: This isn't necessary as there are no modules outside the core library that this project uses.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -19,13 +18,22 @@ jobs:
...
           go get golang.org/x/tools/cmd/cover
           go get github.com/mattn/goveralls
+          go install -v github.com/mattn/goveralls
```
Comment (nicklaw5) [inline .github/workflows/ci.yml:29]: This is also not not needed. The `go get` command install the package.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (jbpratt) [reply .github/workflows/ci.yml:26]: Please see https://github.com/mattn/goveralls/issues/140 but yes I agree, was just hacking this together to get it to work initially.

**Comment by nicklaw5** [comment]
> We can try ripping that stuff out and see what happens

Not sure what you mean by this?

**Comment by jbpratt** [comment]
> > We can try ripping that stuff out and see what happens
> 
> Not sure what you mean by this?

Being inside of gopath during the run

**Comment by nicklaw5** [comment]
> Being inside of gopath during the run

Sorry, I don't understand. Are we making that change to this PR. Or is that a future change?

**Comment by jbpratt** [comment]
Well removing what I did seems to have broken older versions. 

> > Being inside of gopath during the run
> 
> Sorry, I don't understand. Are we making that change to this PR. Or is that a future change?

> You're workflow now only checks out the master branch on build, rather than the branch associated with the PR

My apologies. I added the checkout in gopath because this repo does not contain the standard `go.mod` file and it was causing the action to fail since 1. we weren't operating in `gopath` 2. it is not a valid module to the tool. Adding a `go.mod` file should solve this problem (at least for the newer versions, I can't confidently speak for `v1.8|v1.9`)

It can be done however you would like, or not at all.

**Comment by nicklaw5** [comment]
Ok let's change

```yml
matrix:
        go: [ '1.14.x', '1.13.x', '1.12.x', '1.11.x', '1.10.x', '1.9.x', '1.8.x' ]
```
to
```yml
matrix:
        go: [ '1.15.x', '1.14.x', '1.13.x', '1.12.x' ]
```

**Comment by jbpratt** [comment]
I think we should add a `go.mod` file to sort out the `gopath` shenanigans. It would simply be the following:

```go.mod
module github.com/nicklaw5/helix

go 1.14
```

**Comment by nicklaw5** [comment]
> I think we should add a go.mod file to sort out the gopath shenanigans.

Yep - I was thinking the same.

**Comment by nicklaw5** [comment]
I think we're good?

**Comment by jbpratt** [comment]
> I think we're good?

I think so! Sorry about the hassle.

**Comment by nicklaw5** [comment]
One last thing. It appears have additional brackets around the build badge:

image

**Review by nicklaw5 — APPROVED** [review]
:rocket:

**Comment by nicklaw5** [comment]
Thanks for contributing :slightly_smiling_face:

**Pull request merged**